platform/kube/inject/inject.go

```diff
@@ -190,6 +193,10 @@ func GetInitializerConfig(kube kubernetes.Interface, namespace, injectConfigName
 		return nil, err
 	}
 
+	if len(c.IncludeNamespaces) >= 1 && len(c.ExcludeNamespaces) >= 1 {
+		return nil, fmt.Errorf("Cannot configure both namespaces and excludeNamespaces")
```

Comment:
This needs to be updated if we decided to use IncludedNamespaces

Reply:
I don't think this is what you want. Further down in `inject_test.go` you specify `IncludeNamespaces: NamespaceAll` and `ExcludeNamespaces: "ibm-system"` which this check disallows. You either need to add an excemption for `NamespaceAll` or define precedence for `include` vs. `exclude` to coexist. Either way will need more tests.

Reply:
my point is do we need include and exclude to coexist? this is like what we did with the prepare_proxy.sh script.. We can pick either include or exclude. Both allow you to achieve the same purpose..With exclude, you will be able to say do not inject initializers for ibm-system, istio-system, kube-system, but inject for everything else. (we simplify this further by automatically excludign istio-system and kube-system). 

With include, you say, do not inject for anything other than these x,y,z namespaces.. Having both include and exclude leads to confusion and more mechanisms like precedence, etc. Its not like we support regex etc. in the namespace. Even then, I don't see a use case where you want to have both include and exclude in same configuration.

Reply:
I agree it is confusing if we name it to be IncludedNamespace & ExcludedNamespace.   I would much prefer to have the current namespaces and adding IgnoredNamespace (or ExcludedNamespace) which overwrite what is in the namespaces... as a lot of times, I would not know the namespaces ahead of time as an admin of the cluster and I would only exclude certain system ones that admin have in control.

Reply:
>  You either need to add an exemption for NamespaceAll or ...

My point is the code as-is won't work. A namespace (or set of namespaces) needs to be set, otherwise nothing will be injected. [Initializer.initialize](https://github.com/istio/pilot/blob/8b06fc3a77efd6a78f7bdb285a4f3c00cef810b1/platform/kube/inject/initializer.go#L172) will end up skipping the namespae and the [Wath/List](https://github.com/istio/pilot/blob/master/platform/kube/inject/initializer.go#L114) code won't watch any namespaces.

Simplifying the logic here is fine, but additional code changes are needed to watch/handle `NamespaceAll` when excludeNamespace is used.